app/src/main/java/com/nio/refactor/support/Buffer.java

```diff
@@ -0,0 +1,31 @@
+package com.nio.refactor.support;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+//import com.sun.tools.javac.util.ByteBuffer;
+
+import java.nio.channels.SocketChannel;
+
+public class Buffer {
```

Comment:
I did not get responsibility of the Buffer. If it is Decorator it is ok, but this class is not decorated the buffer. This class is used the buffer to do manipulation with the content and write it is to the client.
SRP is violated